### ISSUE TITLE
Control Module: Drop and prevent reconnect

### ISSUE THREAD

**Issue by topherlandry** [sergey-dryabzhinsky/nginx-rtmp-module#225]
I don't believe there's a directive for this that I can find, and I'm not sure if RTMP supports it, but is there a way to tell the control module to drop a publisher, and also to tell the publisher not to reconnect?

I find when I drop a publisher, if they have auto-reconnect enabled then they just try to reconnect to the server again and again.

**Comment by TimIsOverpowered** [comment]
use the on_publish directive.

**Comment by topherlandry** [comment]
I use `on_publish` already to reject stream keys I have not provided my users. I use the control module to drop a publisher under a different circumstance and that doesn't exactly fit my use case.

I just want the broadcaster to stop trying to reconnect if configured that way.

**Comment by TimIsOverpowered** [comment]
ban the stream key when you drop. then add a check if a stream key is banned. if it is, return 403, if not, return 200.

**Comment by topherlandry** [comment]
Yes, I'm aware of that. But in this case I do not want to ban the stream key. The desired behaviour is for the publisher to be disconnected and remain disconnected, but be able to connect at another time in the future as the key is still active and available.

I am allowing my users to terminate their streams via a button on a control panel. It terminates the stream fine, and yes, works perfectly in the case where the stream key will not be used again. But in the case where the stream key will be used again, the auto-reconnect feature gets in the way.
